EESSI-pilot-install-software.sh

```diff
@@ -62,14 +62,14 @@ export EPREFIX=${EESSI_PREFIX}/compat/${EESSI_OS_TYPE}/${EESSI_CPU_FAMILY}
 DETECTION_PARAMETERS=''
 GENERIC=0
 EB='eb'
+EASYBUILD='EasyBuild'
 if [[ "$1" == "--generic" || "$EASYBUILD_OPTARCH" == "GENERIC" ]]; then
     echo_yellow ">> GENERIC build requested, taking appropriate measures!"
     DETECTION_PARAMETERS="$DETECTION_PARAMETERS --generic"
     GENERIC=1
     EB='eb --optarch=GENERIC'
 fi
 
-
 # make sure we're in Prefix environment by which path to 'bash' command
```

Comment:
Is this check really good enough? My impression is that `$EPREFIX/startprefix` does more than just change your `$PATH`. To really check we should check what they do:
```
if [[ ${SHELL#/cvmfs/pilot.eessi-hpc.org/2021.03/compat/linux/x86_64} != ${SHELL} ]]
then
        echo "You appear to be in prefix already (SHELL=${SHELL})" > /dev/stderr
        exit -1
elif [[ ${SHELL#/cvmfs/pilot.eessi-hpc.org/2021.03/compat/linux/x86_64} != ${SHELL} ]] &&
         [[ ${EPREFIX-unset} == '/cvmfs/pilot.eessi-hpc.org/2021.03/compat/linux/x86_64' ]]
then
        echo "You appear to be in stacked prefix already (EPREFIX=${EPREFIX})" > /dev/stderr
        exit -1
fi
```
This also makes the shebang of this file wrong, it should be ` #!/usr/bin/env bash`